sdk/core/azure-core/inc/azure/core/http/body_stream.hpp

```diff
@@ -68,7 +68,10 @@ namespace Azure { namespace Core { namespace Http {
      */
     virtual void Rewind()
     {
-      throw "Not Implemented"; // TODO: Replace with best practice as defined by guideline
+      // Exception is not meant to be caught. This is rather an indication of a bad program that
+      // needs to be updated. This is the reason of using `throw` alone.
+      throw "The upload BodyStream doesn't support Rewind which is required to guarantee fault "
+            "tolerance when retrying the operation.";
```

Comment:
This exception message (even with my suggestion below) seems very specific to a particular usage of `BodyStream` tied specifically to our pipeline and retry. Couldn't folks use it for other purposes?

```suggestion
      throw "This BodyStream doesn't support Rewind which is required to guarantee fault "
            "tolerance when retrying the operation.";
```

Reply:
See: https://github.com/Azure/azure-sdk-for-cpp/issues/1093#issuecomment-748329982

Reply:
I did already, I don't see how that comment leads to the code change here as the conclusion.